WalletWasabi.Fluent/CrashReport/CrashReporter.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Diagnostics;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Fluent.CrashReport
+{
+	public class CrashReporter
+	{
+		private const int MaxRecursiveCalls = 5;
+		public int Attempts { get; private set; }
+		public string? Base64ExceptionString { get; private set; } = null;
+		public bool HadException { get; private set; }
+		public SerializableException? SerializedException { get; private set; }
+
+		public void TryInvokeIfRequired()
+		{
+			if (!HadException)
+			{
+				return;
+			}
+
+			try
+			{
+				if (Attempts >= MaxRecursiveCalls)
+				{
+					throw new InvalidOperationException($"The crash report has been called {MaxRecursiveCalls} times. Will not continue to avoid recursion errors.");
+				}
+
+				var args = ToCliArguments();
+
+				var path = Process.GetCurrentProcess()?.MainModule?.FileName;
```

Comment:
`Process.GetCurrentProcess()` does not seem to return `null`: https://source.dot.net/#System.Diagnostics.Process/System/Diagnostics/Process.cs,a106ad886675f6a5

Reply:
ACK